File: org.palladiosimulator.blockchainsystems.core/src/main/java/org/palladiosimulator/blockchainsystems/core/mining/MiningProcessFactoryImpl.java
package org.palladiosimulator.blockchainsystems.core.mining;

import java.util.random.RandomGenerator;

import org.palladiosimulator.blockchainsystems.core.system.abstractions.MiningProcess;
import org.palladiosimulator.blockchainsystems.core.system.abstractions.MiningProcessFactory;

public class MiningProcessFactoryImpl implements MiningProcessFactory {

	@Override
	public MiningProcess createMiningProcess(String nodeId) {
		return new MiningProcessImpl(5_000_000, RandomGenerator.of("Random"));
	}

}

File: org.palladiosimulator.blockchainsystems.core/src/main/kotlin/org/palladiosimulator/blockchainsystems/core/mining/MiningProcessFactoryImpl.kt
package org.palladiosimulator.blockchainsystems.core.mining

import org.palladiosimulator.blockchainsystems.core.system.abstractions.MiningProcess
import org.palladiosimulator.blockchainsystems.core.system.abstractions.MiningProcessFactory
import java.util.random.RandomGenerator

class MiningProcessFactoryImpl(
  private val meanBlockTime: Double,
  private val randomGenerator: RandomGenerator
) : MiningProcessFactory {
  override fun createMiningProcess(nodeId: String): MiningProcess {
    return MiningProcessImpl(meanBlockTime, randomGenerator)
  }
}